ZenPacks/zenoss/LinuxMonitor/parsers/linux/cpu.py

```diff
@@ -35,41 +35,55 @@ def processResults(self, cmd, result):
                    'ssCpuIdlePerCpu',
                    'ssCpuUsedPerCpu',
                    'ssCpuWaitPerCpu']
-            
+
+            # New ids for Device CPU
+            userCpuPerCpu = None
+            systemCpuPerCpu = None
+            idleCpuPerCpu = None
+            usedCpuPerCpu = None
+            waitCpuPerCpu = None
+
+            # Component cpu returns socket value, device cpu return empty str
+            component = getattr(cmd, 'component', None)
+
             # Calculate the number of CPUs
             cpuCount=0
+            single_cpu_line = ''
             for line in cmd.result.output.splitlines()[1:]:
               words = line.split()[0:]
               if 'cpu' in words[0]:
                 cpuCount+=1
+                if component:
+                  if component == line[3]:
```

Comment:
Two consecutive if's can be combines.

Reply:
**This code completely fails for systems with more than 10 CPUs.**!

Use words[0][3:]  to capture.